presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PlanNodeDecorrelator.java

```diff
@@ -371,11 +368,11 @@ public PlanNodeDecorrelator(SymbolAllocator symbolAllocator, Lookup lookup)
         public Optional<DecorrelationResult> visitAggregation(AggregationNode node, Void context)
         {
             if (node.hasEmptyGroupingSet()) {
-                return Optional.empty();
+                return visitPlan(node, null);
```

Comment:
In my opinion it conveys well the idea that in this case (e.g. empty grouping set) we don't want to decorrelate with a specific method, but use a default method instead.

Reply:
Please use dedicated method. `visitPlan(node, null);` is too vague to what it does.

Reply:
> Please use dedicated method.

That is this is a good idea. Instead of calling `visitPlan` you could have `visitUnsupportedSuqueryPlan` and then `visitPlan` would just call that method.